examples/chef/chef.py

```diff
@@ -521,6 +521,15 @@ def main() -> int:
     # Cluster customization
     #
 
+    if options.do_run_gui or options.do_run_zap:
+        zap_binary_found = shell.run_cmd(
+            f"if [[ -x $(which zap) ]]; then echo \"zap found\"; else echo \"\"; fi", return_cmd_output=True)
+        zap_dev_path_found = os.getenv("ZAP_INSTALL_PATH")
```

Comment:
This is not sufficient - we should supporet all of the ZAP_INSTALL_PATH and ZAP_DEVELOPMENT_PATH. See:

https://github.com/project-chip/connectedhomeip/blob/master/scripts/tools/zap/zap_execution.py#L33

Also, why do we have to have CHEF check this when our build probably gives error messages that should be readable (https://github.com/project-chip/connectedhomeip/blob/master/scripts/tools/zap/zap_execution.py#L83)

I can see this providing an earlier error at chef.py run time, however it sacrifices support for error messages that are more complete and even if done fully, it would fork the logic in case we want to update anything.

Reply:
I believe run_zaptool is intended to detect all variables:

https://github.com/project-chip/connectedhomeip/blob/master/scripts/tools/zap/run_zaptool.sh#L34

Maybe we should improve error messages in that script instead.
generate.py likely uses zap_execution.py and should have clear error messages.

Reply:
As an aside, what artifacts get generated by `generate.py` ? A lot of app codegen is done now at compile time.

Reply:
Could we update the script and see if there is any more need for zzz_generated and equivalent? 

 #23962 deleted basically all codegen files for chef (like all the other apps) and only af-gen-event remained and that is basically empty. Maybe for chef specifically we can delete those files too and create a common empty af-gen-egen.h.

Reply:
- Leaving the ZAP messaging to zap execution itself
- Deleted codegen from chef itself, as this is handled by the SDK now
- Eliminated inclusion of af-gen-event.h fom af-event.cpp